timely/src/dataflow/operators/to_stream.rs

```diff
@@ -56,3 +60,108 @@ impl<T: Timestamp, I: IntoIterator+'static> ToStream<T, I::Item> for I where I::
         })
     }
 }
+
+/// Data and progress events of the native stream.
+pub enum Event<F: IntoIterator, D> {
+    /// Indicates that timestamps have advanced to frontier F
+    Progress(F),
+    /// Indicates that event D happened at time T
+    Message(F::Item, D),
+}
+
+/// Converts to a timely `Stream`.
+pub trait ToStreamAsync<T: Timestamp, D: Data> {
+    /// Converts a [native `Stream`](futures_util::stream::Stream) of [`Event`s](Event) into a [timely
```

Comment:
Now that this doesn't return a `TimelyFuture`, should it just be an implementation of `ToStream` for `Pin<Box<Stream>>`?

Reply:
I believe this will hit the specialisation issues since `ToStream` is already implemented for a generic `I`

Reply:
Ah, and `Stream` implements `Iterator`? I can't actually tell for its docs.

Reply:
`Stream` doesn't implement `Iterator`, that's the problem! But because Rust you can't have both of:

```
impl<I> ToStream for I where I: Iterator {}
impl<I> ToStream for I where I: Stream {}
```

Reply:
Oh sorry I thought `Stream` was a struct for some reason. Nvm!